internal/wal/wal_aof.go

```diff
@@ -65,31 +170,212 @@ func (w *WALAOF) LogCommand(c *cmd.DiceDBCmd) {
 	data, err := proto.Marshal(entry)
 	if err != nil {
 		slog.Warn("failed to serialize command", slog.Any("error", err.Error()))
+		return
 	}
 
-	writeBuf.Reset()
-	writeBuf.Grow(4 + len(data))
-	if binary.Write(&writeBuf, binary.BigEndian, uint32(len(data))) != nil {
-		slog.Warn("failed to write entry length to WAL", slog.Any("error", err.Error()))
+	buf := bufferPool.Get().([]byte)
+	buf = buf[:0]
+	defer bufferPool.Put(buf)
+
+	length := uint32(len(data))
+	totalSize := 4 + len(data)
+
+	if cap(buf) < totalSize {
+		buf = make([]byte, totalSize)
+	} else {
+		buf = buf[:totalSize]
+	}
+
+	binary.BigEndian.PutUint32(buf[0:4], length)
+	copy(buf[4:], data)
+
+	w.mutex.Lock()
+	defer w.mutex.Unlock()
+
+	if w.rotationMode == "size" {
```

Comment:
This check should probably be a part of the rotateLogIfNeeded method (which internally handles whether rotation is based on size or on time)

Reply:
+1